### ISSUE TITLE
assemble_fibermap standardize SURVEY, PROGRAM

### ISSUE THREAD

**Issue by sbailey** [desihub/desispec#1723]
During sv1, the header keywords for SURVEY, FAFLAVOR, and FAPRGRM were still evolving and different files have different sets of keywords.  However, all fuji+guadalupe tiles have FAFLAVOR in fiberassign HDU 0.  Update assemble_fibermap to use this to standardize SURVEY, PROGRAM in the fibermap headers so that downstream code like desi_tsnr_afterburner doesn't have to re-derive it.

`desispec.io.meta.faflavor2program` extracts and standardizes the PROGRAM part.  Similar utility code could identify the SURVEY prefixes.  Add that as a separate function instead of logic buried in assemble_fibermap itself.

Optionally derive what FAPRGRM would have been if it had been included, and propagate that as well.  When SURVEY, FAFLAVOR, and FAPRGRM all exist, then `FAFLAVOR = SURVEY + FAPRGRM.lower()`, and SURVEY is one of "cmx", "sv1", "sv2", "sv3", "main", "special".

**Comment by sbailey** [comment]
For reference: https://desi.lbl.gov/trac/wiki/SurveyOps/TileDesigns

From a data telecon discussion a few months ago, there was interest in updating these in fiberassign files themselves and not just assemble_fibermap.  I can write the code to do these updates, but will wait for @araichoor and @dstndstn updates so that we don't step on each other.

# Itemizing the cases that exist in fiberassign svn:

## TILEID, SURVEY, FAPRGRM, FAFLAVOR in HDU 0

This is the standard case that we are aiming for.  Nothing needs to be updated for these tiles.

## TILEID in HDU 1 but not HDU 0, and has none of the other keywords

Tiles 63501 to 63505 from early commissioning.  Suggested action for HDU 0:
* set TILEID to match HDU 1
* set SURVEY = cmx
* set FAPRGRM = bright
* set FAFLAVOR = cmxbright

These are very unlikely to ever be processed again, but it is just helpful to have standardized keywords so that we don't trip on them when looping over all tiles.

## TILEID in HDU 0 but none of the rest of the keywords

268 commissioning tiles spanning 59026 to 78013, including 80 that are in daily and tiles-specstatus.ecsv with SURVEY=FAPRGRM=FAFLAVOR=unknown, but not in fuji.  `max(LASTNIGHT) = 20200315`, i.e. pre-covid shutdown, but including some M33 (?) tiles that might be resurrected some day.

Suggested action:
* SURVEY=cmx
* FAPRGRM=unknown
* FAFLAVOR=cmxunknown

including "unknown" is a bit unfortunate, but I don't think it is worth chasing all of these down.

## TILEID and FAFLAVOR in HDU 0, but not SURVEY or FAPRGRM

1573 tiles from 80000 to 82795, FAFLAVOR include dith*, sv1*, cmx*.  Suggested action:

```
if FAFLAVOR.startswith(dith):
    SURVEY = cmx
    FAPRGRM = FAFLAVOR
elif FAFLAVOR in (cmxlrgqso, elg):
    SURVEY = sv1  # NOTE: not cmx
    FAPRGRM = FAFLAVOR[3:]
else:
    SURVEY = FAFLAVOR[0:3]
    FAPRGRM = FAFLAVOR[3:]
```

## TILEID, SURVEY, FAFLAVOR in HDU 0 but not FAPRGRM

24 tiles with FAFLAVOR = sv2*.  Suggested action:

* FAPRGRM = FAFLAVOR[3:]

**Comment by sbailey** [comment]
Implemented in PR #1858.  Closing issue.
